chalice/app.py

```diff
@@ -397,6 +414,30 @@ def json_body(self):
                     raise BadRequestError('Error Parsing JSON')
             return self._json_body
 
+    @property
+    def files(self):
+        content_type = self.headers.get('content-type')
+        if content_type.startswith('multipart'):
+            if self._files is None:
+                self._files = self._generate_files(
+                    content_type, self.raw_body
+                )
+            return self._files
+
+    def _generate_files(self, content_type, raw_body):
```

Comment:
Can we rename this to parse files or something similar?  We're not really generating any new files we're just extracting them out from the request body.